File: src/org/kopi/vkopi/lib/ui/vaadin/addons/client/base/VInputButton.java
package org.kopi.vkopi.lib.ui.vaadin.addons.client.base;

import com.google.gwt.dom.client.Document;
import com.google.gwt.dom.client.InputElement;
import com.google.gwt.event.dom.client.ClickHandler;
import com.google.gwt.http.client.Request;
import com.google.gwt.http.client.RequestBuilder;
import com.google.gwt.http.client.RequestCallback;
import com.google.gwt.http.client.Response;
import com.google.gwt.http.client.URL;
import com.google.gwt.user.client.Event;
import com.google.gwt.user.client.ui.ButtonBase;
/**
 * An input element type button that cannot handle icons.
 */
public class VInputButton extends ButtonBase {

  //---------------------------------------------------
  // CONSTRUCTORS
  //---------------------------------------------------
  
  /**
   * Creates a new input button.
   */
  public VInputButton() {
    this(null);
  }
  
  /**
   * Creates a new button widget.
   * @param caption The button caption
   */
  public VInputButton(String caption) {
    super(Document.get().createButtonInputElement());
    if (caption != null) {
      getInputElement().setValue(caption);
    }
    setStyleName(Styles.INPUT_BUTTON);
    sinkEvents(Event.ONCLICK);
    getElement().setAttribute("hideFocus", "true");
    getElement().getStyle().setProperty("outline", "0px");
  }
  
  /**
   * Creates a new button widget.
   * @param caption The button caption.
   * @param clickHandler The click handler.
   */
  public VInputButton(String caption, ClickHandler clickHandler) {
    this(caption);
    addClickHandler(clickHandler);
  }

  //---------------------------------------------------
  // IMPLEMENTATIONS
  //---------------------------------------------------
  
  /**
   * Fires a click for this button.
   */
  public void click() {
    getInputElement().click();
  }
  
  /**
   * Sets the button caption
   * @param caption The button caption.
   */
  public void setCaption(String caption) {
    getInputElement().setValue(caption);
  }
  
  /**
   * Returns the button caption.
   * @return The button caption.
   */
  public String getCaption() {
    return getInputElement().getValue();
  }
  
  /**
   * Returns the input element.
   * @return the input element.
   */
  public InputElement getInputElement() {
    return getElement().cast();
  }
  
  /**
   * Gets the focus on the button.
   */
  public void focus() {
    setFocus(true);
  }
}

File: src/org/kopi/vkopi/lib/ui/vaadinflow/base/VInputButton.kt
/*
 * Copyright (c) 2013-2022 kopiLeft Services SARL, Tunis TN
 * Copyright (c) 1990-2022 kopiRight Managed Solutions GmbH, Wien AT
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
package org.kopi.vkopi.lib.ui.vaadinflow.base

import com.vaadin.flow.component.ClickEvent
import com.vaadin.flow.component.ClickNotifier
import com.vaadin.flow.component.ComponentEventListener
import com.vaadin.flow.component.html.Input
import com.vaadin.flow.server.VaadinService

/**
 * An input element type button that cannot handle icons.
 */
open class VInputButton(caption: String? = null) : Input(), ClickNotifier<VInputButton> {

  //---------------------------------------------------
  // CONSTRUCTORS
  //---------------------------------------------------

  /**
   * Creates a new input button.
   */
  init {
    type = "button"
    if (caption != null) {
      value = caption
    }
    className = Styles.INPUT_BUTTON
    element.setAttribute("hideFocus", "true")
    element.style["outline"] = "0px"

    addClickListener {
      println("ClientRequest Object : $ClientRequest")
      println("ClientRequest.request Object Before affectation : ${ClientRequest.request}")
      ClientRequest.request = VaadinService.getCurrentRequest()
      println("ClientRequest.request Object After affectation : ${ClientRequest.request}")

      println("(VInputButton)Request is Null ? :${ClientRequest.request == null}")
      println("(VInputButton)ClientRequest.getClientHostname() : ${ClientRequest.getClientHostname()}")
      println("(VInputButton)ClientRequest.getClientIp() : ${ClientRequest.getClientIp()}")

    }
  }

  /**
   * Creates a new button component.
   *
   * @param caption The button caption.
   * @param clickListener The click listener.
   */
  constructor(caption: String?, clickListener: ComponentEventListener<ClickEvent<VInputButton?>?>?) : this(caption) {
    addClickListener(clickListener)
  }

  //---------------------------------------------------
  // IMPLEMENTATIONS
  //---------------------------------------------------
  /**
   * Fires a click for this button.
   */
  fun click() {
    fireEvent(ClickEvent<VInputButton>(this))
  }

  /**
   * The button caption
   */
  var caption: String?
    get() = value
    set(caption) {
      value = caption
    }

  /**
   * sets the name of the component.
   *
   * @return The name.
   */
  fun setName(name: String?) {
    element.setAttribute("name", name)
  }
}
